actors/datacap/src/types.rs

```diff
@@ -0,0 +1,93 @@
+use fvm_ipld_encoding::tuple::*;
+use fvm_ipld_encoding::{Cbor, RawBytes};
+use fvm_shared::address::Address;
+use fvm_shared::bigint::{bigint_ser, BigInt};
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct AllowanceParams {
```

Comment:
Have we considered moving the generalized token type parameters into the token library?

Reply:
Yes, they're not quite ready yet